sdk-ext-ui/src/androidTest/java/cash/z/ecc/sdk/ext/ui/regex/ZecRegexTest.kt

```diff
@@ -0,0 +1,62 @@
+package cash.z.ecc.sdk.ext.ui.regex
+
+import androidx.test.filters.MediumTest
+import cash.z.ecc.sdk.ext.ui.R
+import cash.z.ecc.sdk.ext.ui.test.getStringResourceWithArgs
+import cash.z.ecc.sdk.model.MonetarySeparators
+import org.junit.Test
+import kotlin.test.assertFalse
+import kotlin.test.assertNotNull
+import kotlin.test.assertTrue
+
+class ZecRegexTest {
+
+    @Test
+    @MediumTest
+    fun check_regex_validity() {
+        val separators = MonetarySeparators.current()
+        val regexString = getStringResourceWithArgs(
+            R.string.zec_amount_regex_filter,
+            arrayOf(
+                separators.grouping,
+                separators.grouping,
+                separators.decimal,
+                separators.decimal
+            )
+        )
+        assertNotNull(regexString)
+
+        val regexAmountChecker = regexString.toRegex()
+
+        regexAmountChecker.also {
+            assertNotNull(regexAmountChecker)
+            assertTrue(regexAmountChecker.pattern.isNotEmpty())
+        }
+
+        regexAmountChecker.also {
+            // valid inputs
+            assertTrue(it.matches(""))
+            assertTrue(it.matches("123"))
+            assertTrue(it.matches("123${separators.grouping}"))
```

Comment:
This is good for the regex which is responsible for validation.

However this raises a question about how the ultimate conversion in `Zatoshi.Companion.fromZecString` will work.  If the user types in `123,` and hits send is that converted to `123` or `123,000`?   @geffenz do you have thoughts on what we do here?

Reply:
@ccjernigan @geffenz this particular case of `123,` ends up with `123.00000000` after processed by `Zatoshi.Companion.fromZecString`.